scan/scan_common.go

```diff
@@ -0,0 +1,78 @@
+package scan
+
+var (
+	// Network is the default network to use.
+	Network = "tcp"
+	// Short removes long/expensive tests when set.
+	Short bool
+)
+
+// Grade gives a subjective rating of the host's success in a scan.
+type Grade int
+
+const (
+	// Bad describes a host with serious misconfiguration or vulnerability.
+	Bad Grade = iota
+	// Legacy describes a host with non-ideal configuration that maintains support for legacy clients.
+	Legacy
+	// Good describes host performing the expected state-of-the-art.
+	Good
+)
+
+// Output is the result of a Scan, to be stored for potential use by later Scanners.
+type Output interface{}
+
+// Scanner describes a type of scan to perform on a host.
+type Scanner struct {
+	Name        string
+	Description string
+	scan        func(host string) (Grade, Output, error)
+}
+
+// Scan defines the scan to be performed on the given host.
+func (s *Scanner) Scan(host string) (grade Grade, output Output, err error) {
```

Comment:
Looks good. One style issue: For exported function, we keep the return list nameless, so the doc will be consistent with standard GoDoc. Also, it makes each return statement declare exactly what should be returned. It's much easier to maintain afterwards, and reduces potential bugs.